components/TemplateGrid/TemplateGridContainer.vue

```diff
@@ -45,202 +45,142 @@ export default {
-            }, time)();
+        calculateRowsCount() {
+            const { clientHeight } = document.querySelector('.grid-container');
+            const visibleRows = Math.ceil(clientHeight / this.rowsHeight);
+            const totalRows = Math.max(this.gridData.length, visibleRows) + 1;
+            this.$emit('setRowsCount', totalRows);
         },
-        setGhostItemPosition({ column, row }) {
-            if (
-                (this.lastColumn !== column || this.lastRow !== row)
-                && row !== null
-                && column !== null
-            ) {
-                const newRow = this.setRowIfCollapse(row + 0.5);
-                this.lastColumn = column;
-                this.lastRow = newRow || row;
-                // Seems like lastRow is a ghost element index in the array data
-                // - no need to filter for the ghost element when we have it index
-                this.$emit('addItem', {
-                    item: {
-                        id: this.ghostIndex,
-                        column: this.lastColumn,
-                        row: this.lastRow,
-                        ghost: true,
-                    },
+        onDragOver(event) {
+            const { clientX, clientY } = event;
+            const { overRow, collisionRelation } = this.mousePosition;
+            if (this.onDragFirstItem(event)) return false;
+            const tmpCollisionRelation = this.getMouseOverProps(clientX, clientY);
+            const collidingItem = this.collisionsDetect(overRow);
+
+            if (collidingItem !== null && collisionRelation !== tmpCollisionRelation) {
+                this.setGhostItemPosition(this.getCollidingPosition(collidingItem));
+            } else if (collidingItem === null && overRow > this.maxRow) {
+                this.setGhostItemPosition({
+                    column: this.setAllowedColumn(),
+                    row: this.maxRow + this.positionBeetwenRows,
                 });
             }
+            event.preventDefault();
+            return true;
         },
-        getMouseOverProps(clientX, clientY) {
-            const elements = document.elementsFromPoint(clientX, clientY);
-            const shadowItem = elements.find(e => e.classList.contains('shadow-grid-item'));
-            const shadowItemId = shadowItem ? shadowItem.getAttribute('shadow-id') : null;
-            const overRow = shadowItemId ? Math.floor((shadowItemId - 1) / this.columns) : null;
-            const overColumn = shadowItemId ? (shadowItemId - 1) % this.columns : null;
-            const positionOverRow = shadowItemId
-                ? Math.floor(clientY - shadowItem.getBoundingClientRect().top)
-                : null;
-            return {
-                overColumn,
-                overRow,
-                positionOverRow,
-            };
+        onDragLeave(event) {
+            const { fromElement } = event;
+            if (fromElement && fromElement.classList.contains('blur')) {
```

Comment:
it is not going to remove ghost element in each scenario - something might overflow blur - I would rather check the bounds of grid container